modules/flowable-task-service/src/main/resources/org/flowable/task/db/mapping/entity/Task.xml

```diff
@@ -802,23 +804,31 @@
           <!-- if dbIdentityUsed set true in process engine configuration -->
           <if test="userIdForCandidateAndAssignee != null">
             <if test="candidateGroups == null">
-              and (RES.ASSIGNEE_ = #{userIdForCandidateAndAssignee} or (RES.ASSIGNEE_ is null and (I.USER_ID_ = #{userIdForCandidateAndAssignee}
-              or I.GROUP_ID_ IN (select g.GROUP_ID_ from ${prefix}ACT_ID_MEMBERSHIP g where g.USER_ID_ = #{userIdForCandidateAndAssignee} ) ) ) )
+              and (RES.ASSIGNEE_ = #{userIdForCandidateAndAssignee}
+                or (
+                <if test="candidateAssignedToOthers">
```

Comment:
Should this be <if test="!candidateAssignedToOthers"> instead?

Reply:
I assume you meant the missing negation? You are correct, good catch, I updated it.
It seems that that branch of the query is never tested (neither in the new tests nor in the original ones), hence why it slipped through.